ci/cloudbuild/builds/publish-docs.sh

```diff
@@ -48,6 +48,7 @@ doc_args=(
   "-DGOOGLE_CLOUD_CPP_DOXYGEN_CLANG_OPTIONS=-resource-dir=$(clang -print-resource-dir)"
   "-DGOOGLE_CLOUD_CPP_COMMIT_SHA=${COMMIT_SHA}"
   "-DGOOGLE_CLOUD_CPP_ENABLE_CLANG_ABI_COMPAT_17=ON"
+  "-DGOOGLE_CLOUD_CPP_STORAGE_ENABLE_GRPC=ON"
```

Comment:
cc: @coryan 

I am not sure why this is now necessary. You can try removing it for yourself :shrug: 

I think it was failing on not being able to link the `google-cloud-cpp::experimental-storage_grpc_mocks` target. Which made me think it was linking the [fake](https://github.com/googleapis/google-cloud-cpp/blob/cac2123344fb7dfd6f3fb33dd0bfeaa8c3e7096d/google/cloud/storage/google_cloud_cpp_storage_grpc.cmake#L18-L19) `google-cloud-cpp::experimental-storage_grpc`.

---

https://github.com/googleapis/google-cloud-cpp/blob/cac2123344fb7dfd6f3fb33dd0bfeaa8c3e7096d/cmake/GoogleCloudCppFeatures.cmake#L497-L507

This code looks like it should set the option to `ON`, which is what we want.

Reply:
That shows two problems:

- Maybe `storage_grpc_mocks` should exist when the fake `storage_grpc` library is created.
- I think we have made a hash out of `GOOGLE_CLOUD_CPP_ENABLE`.  It is a cache variable, set by the application, and then we reset it to a value (that is not saved in the cache).  Seems like we should use a different variable (`GOOGLE_CLOUD_CPP_ENABLE_DETAILS` or `_OUTPUT`??) for the computed list of enabled components.